llarp/service/endpoint.cpp

```diff
@@ -139,63 +139,52 @@ namespace llarp::service
+          if (not success)
           {
-            std::string name;
-            try
-            {
-              oxenc::bt_dict_consumer btdc{m.body()};
-              name = btdc.require<std::string>("NAME");
-            }
-            catch (...)
-            {
-              log::warning(link_cat, "Failed to parse find name response!");
-              throw;
-            }
-
-            if (auto saddr = service::Address(); saddr.FromString(name))
-            {
-              ptr->SetAuthInfoForEndpoint(saddr, auth);
-              ptr->MarkAddressOutbound(saddr);
+            result_handler(false, "Exit {} not found!"_format(name));
+            return;
+          }
 
-              auto result = ptr->EnsurePathToService(
-                  saddr,
-                  [ptr, name, ranges, result_handler, poker](auto addr, OutboundContext* ctx) {
-                    if (ctx == nullptr)
+          if (auto saddr = service::Address(); saddr.FromString(name_result))
+          {
+            ptr->SetAuthInfoForEndpoint(saddr, auth);
+            ptr->MarkAddressOutbound(saddr);
+
+            auto result = ptr->EnsurePathToService(
+                saddr,
+                [ptr, name, name_result, ranges, result_handler, poker](
+                    auto addr, OutboundContext* ctx) {
+                  if (ctx == nullptr)
+                  {
+                    result_handler(
+                        false, "could not establish flow to {} ({})"_format(name_result, name));
+                    return;
+                  }
+
+                  // make a lambda that sends the reply after doing auth
+                  auto apply_result = [ptr, poker, addr, result_handler, ranges](
+                                          std::string result, bool success) {
+                    if (success)
                     {
-                      result_handler(false, "could not establish flow to {}"_format(name));
-                      return;
-                    }
-
-                    // make a lambda that sends the reply after doing auth
-                    auto apply_result = [ptr, poker, addr, result_handler, ranges](
-                                            std::string result, bool success) {
-                      if (success)
-                      {
-                        for (const auto& range : ranges)
-                          ptr->MapExitRange(range, addr);
+                      for (const auto& range : ranges)
+                        ptr->MapExitRange(range, addr);
 
-                        if (poker)
-                          poker->put_up();
+                      if (poker)
+                        poker->put_up();
 
-                        result_handler(true, result);
-                      }
+                      result_handler(true, result);
+                    }
 
-                      result_handler(false, result);
-                    };
+                    result_handler(false, result);
```

Comment:
This needs to be in an `else` block so it doesn't roll from calling `result_handler` once into the second invocation